reorganize and update documentation (#125)
JuliaImages/juliaimages.github.io: docs/src/index.md

    Your changes will be vetted by developers before becoming permanent, so don't
    worry about whether you might say something wrong.

> This documentation is a collection of two parts: a top-level documentation that guides you into the

Reviewer: Would it be better to say "This documentation is a collection of several parts." and then delete the next couple of sentences, and launch straight into 'The "Tutorials" part...'?

JuliaImages ecosystem, and a list of more specific documentation of sub-packages that gives you more
information of each image processing subfields.

The "Tutorials" part contains a list of tutorials that help you gain better understanding of the JuliaImages

> ecosystems. The ["Packages"](@ref page_packages_index) part contains all related sub-packages in JuliaImages,

Reviewer: ```suggestion
ecosystem. The ["Packages"](@ref page_packages_index) part contains information about specific components (themselves Julia packages) that together comprise JuliaImages
```

> users can get more information on the specific image processing subfields by reading them. The "Demos"

Reviewer: ```suggestion
and address specific subfields of image processing. The "Demos"
```

> part gives you a task-by-task introduction of how you could do your daily work with JuliaImages.

Reviewer: ```suggestion
part gives you demonstrations of how to carry out specific tasks with JuliaImages.
```

> The "References" part is a collection of function references, it's a dictionary of functions in JuliaImages

Reviewer: ```suggestion
The "References" part is a collection of function references. ```

> and the recommended way to use it is by the searching function of your browser `Ctrl-F`/`Cmd-F`.

Reviewer: ```suggestion
The recommended way to use it is by the searching function of your browser `Ctrl-F`/`Cmd-F`.
```

This documentation is generated with the following environment setup. While reading the documentation,
if you encounter any errors or if the outputs in your local machine differ from the documentation,
you could first check the Julia and package versions you're using. If the error or inconsistency still exists,
please [file an issue](https://github.com/JuliaImages/juliaimages.github.io/issues/new) for that; it
helps us improve the documentation.

```@setup versions
using InteractiveUtils
```
```@repl versions
using Pkg, Dates
today()
versioninfo()

> Pkg.status()

Reviewer 2: Is this really what we need for the top page? I think the top page should be a "welcome" page, as @timholy implies in https://github.com/JuliaImages/juliaimages.github.io/pull/125#discussion_r412076360.

Reviewer 2: Another option is using

    ```@raw html
    <details>
      <summary>Package Status</summary>
    ```
and

    ```@raw html
    </details>
    ```
I don't really recommend it, though,

Author: I believe it is necessary to share the build information somewhere in the documentation. I've now moved this to the troubleshooting section in `install.md`.

> we can find the date and the version from the top-right icon.

Yes but I feel putting them together more convenient to get relevant information.

    Your changes will be vetted by developers before becoming permanent, so don't
    worry about whether you might say something wrong.

> The following is a list of packages that are closely related to the JuliaImages ecosystem and also maintained by members of

Reviewer: I like the idea here, but I worry this list will be intimidating for newcomers ("Welcome to JuliaImages! Here are 60 different packages you have to master. Good luck!"), especially if it appears on the front page. I'd rather start with a gentle introduction before getting into a long list spelling out what code goes where. Let's just do `using Images` until readers gain a bit of experience with JuliaImages.

Author: This is moved to the front page of the "Pakcages" part. And I've also added some words to explain how the documentation is organized and built.
